### DISCUSSION TITLE
install process for expo is bad

### DISCUSSION THREAD

**Discussion by kjoedion** [nativewind/nativewind#337] [General]
i just want to say that i really prefer the nativewind DX (e.g. using `className` like regular react) but getting it running for expo web sucks

i was [following the guide](https://www.nativewind.dev/quick-starts/expo) and got to the part where it links to a demo repo, and just trying to sift through what packages i need to install, what files i need to update, etc. is not straight forward at all

compare this to `npm i twrnc` where it just works immediately with a single import

if you had a better install process/docs for web i think this package would easily overtake [twrnc in npmtrends](https://npmtrends.com/nativewind-vs-react-native-tailwindcss-vs-tailwind-rn-vs-twrnc) but currently the install process is too painful so i will stick with that package for now, despite the DX being a tiny bit worse

**Comment by marklawlor** [comment]
Hey! Yes the installation for Expo is painful :cry: Unfortunately this the because Expo Web has not been updated for a while and has fallen out-of-date. Unlike other web frameworks it does not provide first-class support for `postcss` (what Tailwind uses under-the-hood) and has not yet updated from Webpack 4 to Webpack 5. 

 I've spoken to the Expo team about this and hopefully I can work with them on improving Expo Web. Most web users I've spoken to have stopped using Expo Web and are now using the Expo Next.js Adapter (which properly supports Tailwind). The other option is not to use Expo Web and follow the React Native Web setup guide with a custom config https://necolas.github.io/react-native-web/docs/installation/

Unfortunately I can't force Expo to update their tooling - so it's going to remain painful until then.

If `twrnc` "just works" for you - that's great. But NativeWind has been successful because of `twrnc`'s short-comings. While the two libraries may appear similar, they are fundamentally different (NativeWind directly uses TailwindCSS while twnrc uses a custom compiler that behaves like TailwindCSS).

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
